### PR TITLE
Revert k8s126 update

### PR DESCRIPTION
**Pull request by tmjd**
## Description

The update to K8s v1.26 pins has broken our ability to wait to establish watches after the apiserver comes up. I've reverted the [commit](https://github.com/tigera/operator/pull/2586/commits/a3b32bc63a48057586f6aab5e793f7421e681419) (originally part of #2586) and added back a change that was in that commit to strip out symbols from the built binary.

## For PR author

- [ ] Tests for change.
- [ ] If changing pkg/apis/, run `make gen-files`
- [ ] If changing versions, run `make gen-versions`

## For PR reviewers

A note for code reviewers - all pull requests must have the following:

- [ ] Milestone set according to targeted release.
- [ ] Appropriate labels:
  - `kind/bug` if this is a bugfix.
  - `kind/enhancement` if this is a a new feature.
  - `enterprise` if this PR applies to Calico Enterprise only.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -265,7 +265,7 @@ endif
 BINDIR?=build/init/bin
 $(BINDIR)/kubectl:
 	mkdir -p $(BINDIR)
-	curl -L https://storage.googleapis.com/kubernetes-release/release/v1.26.3/bin/linux/$(ARCH)/kubectl -o $@
+	curl -L https://storage.googleapis.com/kubernetes-release/release/v1.25.6/bin/linux/$(ARCH)/kubectl -o $@
```
Comment (rene-dekker) [inline Makefile:268]: nit: Do we want do use also 1.25.10 here?

File: go.mod [hunk 2]
```diff
@@ -22,21 +22,22 @@ require (
...
 	gopkg.in/inf.v0 v0.9.1
 	gopkg.in/yaml.v2 v2.4.0
-	k8s.io/api v0.26.3
-	k8s.io/apiextensions-apiserver v0.26.3
-	k8s.io/apimachinery v0.26.3
-	k8s.io/client-go v0.26.3
-	k8s.io/kube-aggregator v0.26.3
-	k8s.io/utils v0.0.0-20230313181309-38a27ef9d749
-	sigs.k8s.io/controller-runtime v0.14.6
+	k8s.io/api v0.25.8
```
Comment (rene-dekker) [inline go.mod:29]: There is now also:
```
	k8s.io/api v0.25.10
	k8s.io/apiextensions-apiserver v0.25.10
	k8s.io/apimachinery v0.25.10
	k8s.io/client-go v0.25.10
	k8s.io/kube-aggregator v0.25.10
```

**Review by rene-dekker — APPROVED** [review]
lgtm

**Comment by tmjd** [comment]
This isn't needed to master. I need to update my other PR with a CRD update because it changed k8s versions.

**Comment by tmjd** [comment]
I closed the wrong PR

File: go.mod (hunk 2 above)
Comment (tmjd) [reply go.mod:29]: Lets do this in a follow up one, I don't want to hold this up any.

**Pull request merged**